json-endpoints/class.wpcom-json-api-get-site-endpoint.php

```diff
@@ -54,430 +119,273 @@ function callback( $path = '', $blog_id = 0 ) {
+		foreach ( $options_response_keys as $key ) {
+			switch ( $key ) {
+				case 'timezone' :
+					$options[ $key ] = (string) get_option( 'timezone_string' );
+					break;
+				case 'gmt_offset' :
+					$options[ $key ] = (float) get_option( 'gmt_offset' );
+					break;
+				case 'videopress_enabled' :
+					$options[ $key ] = $this->site->has_videopress();
+					break;
+				case 'upgraded_filetypes_enabled' :
+					$options[ $key ] = $this->site->upgraded_filetypes_enabled();
+					break;
+				case 'login_url' :
+					$options[ $key ] = wp_login_url();
+					break;
+				case 'admin_url' :
+					$options[ $key ] = get_admin_url();
+					break;
+				case 'is_mapped_domain' :
+					$options[ $key ] = $this->site->is_mapped_domain();
+					break;
+				case 'is_redirect' :
+					$options[ $key ] = $this->site->is_redirect();
+					break;
+				case 'unmapped_url' :
+					$options[ $key ] = get_site_url( $this->site->blog_id );
+					break;
+				case 'featured_images_enabled' :
+					$options[ $key ] = $this->site->featured_images_enabled();
+					break;
+				case 'theme_slug' :
+					$options[ $key ] = get_option( 'stylesheet' );
+					break;
+				case 'header_image' :
+					$options[ $key ] = get_theme_mod( 'header_image_data' );
+					break;
+				case 'background_color' :
+					$options[ $key ] = get_theme_mod( 'background_color' );
+					break;
+				case 'image_default_link_type' :
+					$options[ $key ] = get_option( 'image_default_link_type' );
+					break;
+				case 'image_thumbnail_width' :
+					$options[ $key ] = (int) get_option( 'thumbnail_size_w' );
+					break;
+				case 'image_thumbnail_height' :
+					$options[ $key ] = (int) get_option( 'thumbnail_size_h' );
+					break;
+				case 'image_thumbnail_crop' :
+					$options[ $key ] = get_option( 'thumbnail_crop' );
+					break;
+				case 'image_medium_width' :
+					$options[ $key ] = (int) get_option( 'medium_size_w' );
+					break;
+				case 'image_medium_height' :
+					$options[ $key ] = (int) get_option( 'medium_size_h' );
+					break;
+				case 'image_large_width' :
+					$options[ $key ] = (int) get_option( 'large_size_w' );
+					break;
+				case 'image_large_height' :
+					$options[ $key ] = (int) get_option( 'large_size_h' );
+					break;
+				case 'permalink_structure' :
+					$options[ $key ] = get_option( 'permalink_structure' );
+					break;
+				case 'post_formats' :
+					$options[ $key ] = $this->site->get_post_formats();
+					break;
+				case 'default_post_format' :
+					$options[ $key ] = get_option( 'default_post_format' );
+					break;
+				case 'default_category' :
+					$options[ $key ] = (int) get_option( 'default_category' );
+					break;
+				case 'allowed_file_types' :
+					$options[ $key ] = $this->site->allowed_file_types();
+					break;
+				case 'show_on_front' :
+					$options[ $key ] = get_option( 'show_on_front' );
+					break;
+				/** This filter is documented in modules/likes.php */
```

Comment:
You'll need to move the comment right above the filter declaration so the parser can see it.